src/commands/test/lib/junit-xml-util.ts

```diff
@@ -12,46 +11,26 @@ export class JUnitXmlUtil extends XmlUtil {
     let mainXml: Document = this.getEmptyXmlDocument();
 
     const self: JUnitXmlUtil = this;
-    return new Promise<Document>((resolve, reject) => {
-      fs.createReadStream(pathToArchive)
-        .pipe(unzip.Parse())
-        .on("entry", function (entry: unzip.Entry) {
-          // Skip directories and hidden system files
-          if (entry.type === "Directory" || path.basename(entry.path).substring(0, 1) === ".") {
-            return;
-          }
-          const fullPath: string = path.join(tempPath, entry.path);
-          entry.pipe(fs.createWriteStream(fullPath).on("close", () => {
-            try {
-              // Handle DOMParser warnings, errors and fatalErrors like JS exceptions
-              const configuration: object = {
-                locator: {},
-                errorHandler: function (level: string, msg: string) {
-                  throw `DOMParser${level}: ${msg}`;
-                }
-              };
-
-              const xml: Document = new DOMParser(configuration).parseFromString(fs.readFileSync(fullPath, "utf-8"), "text/xml");
-
-              let name: string = "unknown";
-              const matches: RegExpMatchArray = entry.path.match("^(.*)_TEST.*");
-              if (matches && matches.length > 1) {
-                name = matches[1].replace(/\./gi, "_");
-              }
-
-              self.appendToTestNameTransformation(xml, name);
-              self.removeIgnoredTransformation(xml);
-
-              mainXml = self.combine(mainXml, xml);
-            } catch (e) {
-              reject(e);
-            }
-          }));
-        })
-        .on("close", () => {
-          resolve(mainXml);
-        });
-    });
+
+    return this.getMergeXmlResultsPromise(pathToArchive, tempPath,
+      (fullPath: string, entry: unzip.Entry) => {
+        const xml: Document = new DOMParser(XmlUtil.DOMParserConfig).parseFromString(fs.readFileSync(fullPath, "utf-8"), "text/xml");
+
+        let name: string = "unknown";
+        const matches: RegExpMatchArray = entry.path.match("^(.*)_TEST.*");
+        if (matches && matches.length > 1) {
+          name = matches[1].replace(/\./gi, "_");
+        }
+
+        self.appendToTestNameTransformation(xml, name);
+        self.removeIgnoredTransformation(xml);
+
+        mainXml = self.combine(mainXml, xml);
```

Comment:
Just noticed this - should this be the same as the NUnit one?

```
if (mainXml) {
          mainXml = self.combine(mainXml, xml);
        } else {
          mainXml = xml;
     }
```

Reply:
If we completely merge these sections between NUNit and JUnit - we could use this logic though? Because `mainXml` will only return `false` for JUnit?

Reply:
We have different regex as well and use of `self.removeEmptySuitesTransformation(xml);` for nunit. In addition, implementation and usage of appendToTestNameTransformation and removeIgnoredTransformation is different, so we need multiple parameters for the shared function. 
For me, when number of parameters for some shared code is large - it's a reason to not reuse everything 😃

Reply:
Of course you're correct! I hadn't noticed the difference in the regex.